test.js

```diff
@@ -200,6 +200,28 @@ test('should deserialize plain object', t => {
 	t.is(deserialized.code, 'code');
 });
 
+test('should deserialize nested errors', t => {
```

Comment:
I duplicate this test because the non-enumerable `cause` is treated differently from other props, so this test can work while `cause` fails and viceversa

Reply:
I inlined this comment and improved the code